geopy

calellowitz opened the pull request:
I am not super happy with this code yet, but this is the basic approach I took. In the map view, it is green for the form being considered, blue for nearby forms from the same user, and red (none in this shot) for forms from other users.

Planning to adjust the padding on the buttons and add the tooltips tomorrow.

![image](https://github.com/dimagi/commcare-connect/assets/6844721/09dde63e-cb11-4a1c-9787-fc919529c7b2)

pxwxnvermx commented on commcare_connect/opportunity/views.py:
```diff
@@ -497,10 +498,38 @@ def visit_verification(request, org_slug=None, pk=None):
+                user_forms.append((loc, dist.m, other_lat, other_lon))
+            else:
+                other_forms.append((loc, dist.m, other_lat, other_lon))
+            user_forms.sort(key=lambda x: x[1])
```
is this sorting for forms necessary for the UI?

calellowitz replied:
yeah, we only want the 5 closest forms (you can see when we pass the list to the page that we only take the first five), so we sort them by distance here

ctsims replied:
@calellowitz  - Note that we want the 5 closest forms within some threshold. I'm going to arbitrarily say that the threshold is 250 meters based on heuristically spot checking the data around. 

@pxwxnvermx one other thing I'm going to ask for at some point (possibly while Cal is offline) is the ability to change the actual marker images to be different (not just color) for the two types of geopoint, and to add a Legend to either the map view or next to the map view. If either of those is relevant for Cal's code here, good to plan for ahead.

calellowitz replied:
> Note that we want the 5 closest forms within some threshold. I'm going to arbitrarily say that the threshold is 250 meters based on heuristically spot checking the data around.

Updated to cap at 250 meters.

The other two will require a bit more work, but should be achievable. I can also pick up one or both on monday when I get back.

pxwxnvermx reviewed (approved):
Just a query. Rest looks good.

calellowitz commented:
Made a few updates to the UI (popups, moved the buttons, fixed the duration). Going to leave it like this for now, but should be good to merge as is, or can incorporate more features

![image](https://github.com/dimagi/commcare-connect/assets/6844721/52d97d95-ed8e-44c6-b2d6-3efe9688a35f)
